config/environments/staging.rb

```diff
@@ -0,0 +1,94 @@
+
+  # Disable request forgery protection in test environment
+  config.action_controller.allow_forgery_protection = false
+
+  # Tell Action Mailer not to deliver emails to the real world.
+  # The :test delivery method accumulates sent emails in the
+  # ActionMailer::Base.deliveries array.
+  config.action_mailer.delivery_method = :test
+
+  # Don't care if the mailer can't send
+  config.action_mailer.raise_delivery_errors = false
+
+  # Print deprecation notices to the Rails logger
+  config.active_support.deprecation = :log
+
+  # Only use best-standards-support built into browsers
+  config.action_dispatch.best_standards_support = :builtin
+
+  # Silences warning
+  config.eager_load = false
+
+  require 'rack-webauth/test'
+  if Settings.WEBAUTH
+    config.middleware.use(
+      Rack::Webauth::Test,
+      user: Settings.WEBAUTH.USER,
+      mail: Settings.WEBAUTH.MAIL,
+      ldapprivgroup: Settings.WEBAUTH.LDAPPRIVGROUP,
+      suaffiliation: Settings.WEBAUTH.SUAFFILIATION,
+      displayname: Settings.WEBAUTH.DISPLAYNAME,
+      ldapauthrule: Settings.WEBAUTH.LDAPAUTHRULE
+    )
+  end
+  config.middleware.use(Rack::Webauth)
+
+  if Settings.APPLICATION.ASSETS.PRECOMPILE
+    config.assets.precompile += Settings.APPLICATION.ASSETS.PRECOMPILE
+  end
+
+  Argo.configure do
+    reindex_on_the_fly      Settings.REINDEX_ON_FLY
+    date_format_str         Settings.DATE_FORMAT_STR
+    urls do
+      mdtoolkit             Settings.MDTOOLKIT_URL
+      purl                  Settings.PURL_URL
+      stacks                Settings.STACKS_URL
+      stacks_file           Settings.STACKS_FILE_URL
+      dor_services          Settings.DOR_SERVICES_URL
+      workflow              Settings.WORKFLOW_URL
+      dpg                   Settings.DPG_URL
+      robot_status          Settings.ROBOT_STATUS_URL
+      modsulator            Settings.MODSULATOR_URL
+      normalizer            Settings.NORMALIZER_URL
+      spreadsheet           Settings.SPREADSHEET_URL
+    end
+    bulk_metadata_directory Settings.BULK_METADATA.DIRECTORY # Directory for storing bulk metadata job info
+    bulk_metadata_log       Settings.BULK_METADATA.LOG       # Bulk metadata log file
+    bulk_metadata_csv_log   Settings.BULK_METADATA.CSV_LOG   # Bulk metadata log file in CSV format for end users
+    bulk_metadata_xml       Settings.BULK_METADATA.XML       # Bulk metadata XML output file
+  end
+
+  # the following config statement gets us two important things on each log line:
+  # 1) unique IDs per request, so that it's easier to correlate
+  # logging statements associated with a given request when multiple
+  # simultaneous requests have interleaved log statements.
+  # 2) time stamps on each log statement, because knowing when something
+  # got logged is quite nice.
+  # (declared after Argo.configure since it uses one of those params)
+  config.log_tags = [:uuid, proc {DateTime.now.strftime(Argo::Config.date_format_str)}]
```

Comment:
Do not use `DateTime.now.strftime` without zone. Use one of `Time.zone.now.strftime`, `DateTime.now.strftime.current`, `DateTime.now.strftime.in_time_zone`, `DateTime.now.strftime.utc`, `DateTime.now.strftime.getlocal`, `DateTime.now.strftime.iso8601`, `DateTime.now.strftime.jisx0301`, `DateTime.now.strftime.rfc3339`, `DateTime.now.strftime.to_i`, `DateTime.now.strftime.to_f` instead.